### ISSUE TITLE
Misleading "Localization / Cookies" section

### ISSUE THREAD

**Issue by PaloMraz** [aspnet/AspNetCore.Docs#15371]
The "Localization / Cookies" section is misleading – out of nowhere, it mentions HostModel and its OnGet method in Pages/Host.cshtml.cs file, which does not exist (at least not in a project created from the blazorserver template). I believe that HostModel is a Razor pages project feature; blazorserver projects have Pages/_Host.cshtml hosting page. Therefore, the correct instruction to enable custom localization scheme using the CookieRequestCultureProvider is to add the cookie-setting code to the Pages/_Host.cshtml like this:

&lt;body&gt;
  @{
    this.HttpContext.Response.Cookies.Append(
      Microsoft.AspNetCore.Localization.CookieRequestCultureProvider.DefaultCookieName,
      Microsoft.AspNetCore.Localization.CookieRequestCultureProvider.MakeCookieValue(
        new Microsoft.AspNetCore.Localization.RequestCulture(
            System.Globalization.CultureInfo.CurrentCulture,
            System.Globalization.CultureInfo.CurrentUICulture)));
  }
  &lt;app&gt;
    @(await Html.RenderComponentAsync&lt;App&gt;(RenderMode.ServerPrerendered))
  &lt;/app&gt;
  &lt;script src="_framework/blazor.server.js"&gt;&lt;/script&gt;
&lt;/body&gt;

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: be4c4d0c-149c-bde1-c7fa-f082916390ca
* Version Independent ID: 9d945996-ad02-bd4b-58ea-c465cc5cc74c
* Content: [Create and use ASP.NET Core Razor components](https://docs.microsoft.com/en-us/aspnet/core/blazor/components?view=aspnetcore-3.0#cookies)
* Content Source: [aspnetcore/blazor/components.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/blazor/components.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-blazor**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

**Comment by guardrex** [comment]
I'll cross-link the work first ...

PR was https://github.com/aspnet/AspNetCore.Docs/pull/13724
Issue (includes the source content): https://github.com/aspnet/AspNetCore.Docs/issues/13436

... and ask @rynowak for feedback on your remarks.

**Comment by Stamo-Gochev** [comment]
Can you also add some info how the section "Providing UI to choose the culture" from https://github.com/aspnet/AspNetCore.Docs/issues/13436 should be translated to the template blazor project? There are no controllers there.

**Comment by guardrex** [comment]
@Stamo-Gochev There is a controller: `CultureController` with an action `SetCulture` receiving the request from the component's `OnSelected`: `UriHelper.NavigateTo("/Culture/SetCulture" + query, forceLoad: true);`.

**Comment by Stamo-Gochev** [comment]
@guardrex Controllers in general are not part of the template blazor project. The examples I found are all about how to make localization work in ASP.NET Core MVC project - and they work fine. But what about blazor apps? Can you add an example that makes this work in a server-side blazor app? What about wasm and the problem with `.resx` files there?

**Comment by guardrex** [comment]
It's covered in the MVC area of the docs.

Add to the service collection (`Startup.ConfigureServices`) ...

```csharp
services.AddControllers();
```

Add the controller for endpoint routing in `Startup.Configure` ...

```csharp
app.UseEndpoints(endpoints =>
{
    endpoints.MapControllers();
    endpoints.MapBlazorHub();
    endpoints.MapFallbackToPage("/_Host");
});
```

With the bits in the topic for Blazor Server, binding for loc should work after that according to https://docs.microsoft.com/en-us/aspnet/core/blazor/components?view=aspnetcore-3.0#use-net-localization-scenarios-in-blazor-apps.

I can't remark on the Blazor WebAssembly approach. I was merely a conduit for moving loc content into the doc. I'll :ear: for engineering to provide further guidance on the Blazor WebAssembly scenario.

**Comment by guardrex** [comment]
btw -- now that we've noted here that those MVC bits are missing from coverage, I'll see about cross-linking to surface that. :point_up:

**Comment by Stamo-Gochev** [comment]
@guardrex My concrete case is that I want to localize a blazor component from a custom class library project. I suppose that this should be done by changing the culture and using `IStringLocalizer`, but this does not seem to be working. On the other hand, the examples from the docs show how to localize from a controller.

**Comment by texyh** [comment]
TRUE!!, that docs mislead me for days, thanks for this correction. i have a question tho, so in my case, the user language is stored in the server, please i want to know where is the best place  to make the api call to get the user languages and set the cultureinfo using the language. @PaloMraz  @guardrex

**Comment by GioviQ** [comment]
Documentation and example for blazor webassembly localization are needed.
At the moment using  IStringLocalizer, I have some issue with UI refresh.
